Formula/joern.rb

```diff
@@ -0,0 +1,73 @@
+class Joern < Formula
+  desc "Open-source code analysis platform based on code property graphs"
+  homepage "https://joern.io/"
+  url "https://github.com/joernio/joern/archive/refs/tags/v1.1.1384.tar.gz"
+  sha256 "c71b77c519b051a07728e20eb78538f6c76287e7ec2ee4ee082a8fd852f67dad"
+  license "Apache-2.0"
+
+  depends_on "node@18" => :build
```

Comment:
Does it not work with the latest?

Reply:
It does not work on the latest. I detailed the error that was encountered and link to the failing line in the relevant CI run [in this prior response](https://github.com/Homebrew/homebrew-core/pull/120622#discussion_r1070447453).

Reply:
Can you make an issue upstream then? Because failing to keep up with dependencies is an indication for me that a formula will be impossible to maintain later on.

Reply:
Diving into this, it seems to be a limitation of `pkg` which is being used here for `astgen`. Of course, `astgen` might be moved to a separate formula, but the issue would be encountered there so it may be worth talking about now. It seems `pkg` depends on `pkg-fetch` which tends to support even-versioned node releases per their [Binary Compatibility table](https://github.com/vercel/pkg-fetch#binary-compatibility). Perhaps it would be sufficient to detail this limitation wherever it is encountered with a link to the Binary Compatibility table so that it is easy to maintain?

Reply:
The astgen dependency/resource will be removed from the formula once https://github.com/joernio/joern/pull/2191 is integrated, which in turn will allow me to fully eliminate the dependency on node from this formula.

Reply:
Fixed as of https://github.com/Homebrew/homebrew-core/pull/120622/commits/b637d02b06d0ff043121d1b0ad51a91fc343de41